sql/expression/function/regexp_replace.go

```diff
@@ -0,0 +1,257 @@
+}
+
+// FunctionName implements sql.FunctionExpression
+func (r *RegexpReplace) FunctionName() string {
+	return "regexp_replace"
+}
+
+// Type implements the sql.Expression interface.
+func (r *RegexpReplace) Type() sql.Type { return sql.LongText }
+
+// IsNullable implements the sql.Expression interface.
+func (r *RegexpReplace) IsNullable() bool { return true }
+
+// Children implements the sql.Expression interface.
+func (r *RegexpReplace) Children() []sql.Expression {
+	return r.args
+}
+
+// Resolved implements the sql.Expression interface.
+func (r *RegexpReplace) Resolved() bool {
+	for _, arg := range r.args {
+		if !arg.Resolved() {
+			return false
+		}
+	}
+	return true
+}
+
+// WithChildren implements the sql.Expression interface.
+func (r *RegexpReplace) WithChildren(children ...sql.Expression) (sql.Expression, error) {
+	if len(children) != len(r.args) {
+		return nil, sql.ErrInvalidChildrenNumber.New(r, len(children), len(r.args))
+	}
+	return NewRegexpReplace(children...)
+}
+
+func (r *RegexpReplace) String() string {
+	var args []string
+	for _, e := range r.args {
+		args = append(args, e.String())
+	}
+	return fmt.Sprintf("regexp_replace(%s)", strings.Join(args, ", "))
+}
+
+func (r *RegexpReplace) compile(ctx *sql.Context, row sql.Row) {
+	pattern := r.args[1]
+	var flags sql.Expression = nil
+	if len(r.args) == 6 {
+		flags = r.args[5]
+	}
+	r.re, r.compileErr = compileRegex(ctx, pattern, flags, r.FunctionName(), row)
+}
+
+// Eval implements the sql.Expression interface.
+func (r *RegexpReplace) Eval(ctx *sql.Context, row sql.Row) (interface{}, error) {
+	span, ctx := ctx.Span("function.RegexpReplace")
+	defer span.Finish()
+
+	cached := r.cachedVal.Load()
+	if cached != nil {
+		return cached, nil
+	}
+
+	// Evaluate string value
+	str, err := r.args[0].Eval(ctx, row)
+	if err != nil {
+		return nil, err
+	}
+	if str == nil {
+		return nil, nil
+	}
+	str, err = sql.LongText.Convert(str)
+	if err != nil {
+		return nil, err
+	}
+
+	// Convert to string
+	_str := str.(string)
+
+	// Create regex, should handle null pattern and null flags
+	r.compile(ctx, row)
+	if r.compileErr != nil {
+		return "compile bad", r.compileErr
+	}
+	if r.re == nil {
+		return nil, nil
+	}
+
+	// Evaluate ReplaceStr
+	replaceStr, err := r.args[2].Eval(ctx, row)
+	if err != nil {
+		return nil, err
+	}
+	if replaceStr == nil {
+		return nil, nil
+	}
+	replaceStr, err = sql.LongText.Convert(replaceStr)
+	if err != nil {
+		return nil, err
+	}
+
+	// Convert to string
+	_replaceStr := replaceStr.(string)
+
+	// Do nothing if str is empty
+	if len(_str) == 0 {
+		return _str, nil
+	}
+
+	// Check if position argument was provided
+	var pos interface{} = nil
+	if len(r.args) >= 4 {
+		pos, err = r.args[3].Eval(ctx, row)
+		if err != nil {
+			return nil, err
+		}
+		if pos == nil {
+			return nil, nil
+		}
+	} else {
+		// Default position is 1, if argument not provided
+		pos = 1
+	}
+
+	// Convert to int32
+	pos, err = sql.Int32.Convert(pos)
```

Comment:
And move this code inside the >= 4 block above, no reason to execute it on a `1`